src/main/scala/mesosphere/marathon/core/launcher/impl/InstanceOpFactoryImpl.scala

```diff
@@ -336,23 +443,38 @@ object InstanceOpFactoryImpl {
 
     assume(!hostPorts.flatten.contains(0), "expected that all dynamic host ports have been allocated")
 
-    val since = clock.now()
-
     val allocPortsByCTName: Seq[(String, Int)] = reqPortsByCTName.zip(hostPorts).collect {
       case ((name, Some(_)), Some(allocatedPort)) => name -> allocatedPort
     }(collection.breakOut)
 
+    taskIDs.map { taskId =>
+      // the task level host ports are needed for fine-grained status/reporting later on
+      val taskHostPorts: Seq[Int] = taskId.containerName.map { ctName =>
+        allocPortsByCTName.withFilter { case (name, port) => name == ctName }.map(_._2)
+      }.getOrElse(Seq.empty[Int])
+
+      val networkInfo = NetworkInfo(agentInfo.host, taskHostPorts, ipAddresses = Nil)
+      taskId -> networkInfo
+    }(collection.breakOut)
+  }
```

Comment:
Is this shift of code necessary?

Reply:
What do you mean? A common code got factored out into a separate function, AFAICT.